src/snowflake/connector/cache.py

```diff
@@ -101,15 +112,15 @@ def __setitem(
             expiry=now() + self._entry_lifetime,
             entry=v,
         )
-        self.telemetry["size"] = len(self._cache)
+        self._add_or_remove()
```

Comment:
I'm wondering if it makes sense to have a subclass that handles telemetry? For example, `self._telemetry.add_or_remove()`. This could reduce ambiguity as `_add_or_remove` also sounds like an action.

Reply:
I think the current design is good enough for what I'd want out of cache telemetry. It's important to note that the consumer of this class and its telemetry is us and not our users and this API is not exposed to them, so it can be changed anytime.